packages/core/src/utils.ts

```diff
@@ -72,6 +72,54 @@ export function throttle(
   }
 }
 
+/**
+ * Performs a deep merge of objects and arrays
+ * - arrays values are merged index by index
+ * - objects are merged by keys
+ * - values get replaced, unless undefined
+ *
+ * ⚠️ this method does not prevent infinite loops while merging circular references ⚠️
+ *
+ * @param {destination, ...toMerge} any - values to merge
+ * @returns {any} New merged value
```

Comment:
I would be in favor of not document `params` and `returns` since naming is pretty explicit and we would need to maintain types here too (which seems already outdated).